forte/processors/data_augment/algorithms/skeleton_op.py

```diff
@@ -0,0 +1,765 @@
+        new_pack.add_entry(new_entry)
+        entry_map[entry.tid] = new_entry.tid
+        return True
+
+    def _copy_multi_pack_link_or_group(
+        self, entry: Union[MultiPackLink, MultiPackGroup], multi_pack: MultiPack
+    ) -> bool:
+        r"""
+        This function copies a MultiPackLink/MultiPackGroup in the multipack.
+        It could be used in tasks such as text generation, where
+        MultiPackLink is used to align the source and target.
+        Args:
+            entry: The MultiPackLink/MultiPackGroup to copy.
+            multi_pack: The multi_pack contains the input entry.
+        Returns:
+            A bool value indicating whether the copy happens.
+        """
+        # The entry should be either MultiPackLink or MultiPackGroup.
+        is_link: bool = isinstance(entry, BaseLink)
+        children: List[Entry]
+        if is_link:
+            children = [entry.get_parent(), entry.get_child()]  # type: ignore
+        else:
+            children = entry.get_members()  # type: ignore
+
+        # Get the copied children entries.
+        new_children: List[Entry] = []
+        for child_entry in children:
+            child_pack: DataPack = child_entry.pack
+            child_pack_pid: int = child_pack.pack_id
+            # The new pack should be present.
+            if (
+                child_pack_pid not in self._data_pack_map
+                or child_pack_pid not in self._entry_maps
+            ):
+                return False
+            new_child_pack: DataPack = multi_pack.get_pack_at(
+                multi_pack.get_pack_index(self._data_pack_map[child_pack_pid])
+            )
+            # The new child entry should be present.
+            if child_entry.tid not in self._entry_maps[child_pack_pid]:
+                return False
+            new_child_tid: int = self._entry_maps[child_pack_pid][
+                child_entry.tid
+            ]
+            new_child_entry: Entry = new_child_pack.get_entry(new_child_tid)
+            new_children.append(new_child_entry)
+
+        # Create the new entry and add to the multi pack.
+        new_entry: Entry
+        if is_link:
+            entry = cast(MultiPackLink, entry)
+            new_link_parent: Entry = new_children[0]
+            new_link_child: Entry = new_children[1]
+            new_entry = type(entry)(
+                multi_pack, new_link_parent, new_link_child  # type: ignore
+            )
+        else:
+            entry = cast(MultiPackGroup, entry)
+            new_entry = type(entry)(multi_pack, new_children)  # type: ignore
+        multi_pack.add_entry(new_entry)
+        return True
+
+    def modify_index(
+        self,
+        index: int,
+        # Both of the following spans should be SortedList.
+        # Use List to avoid typing errors.
+        old_spans: List[Span],
+        new_spans: List[Span],
+        is_begin: bool,
+        is_inclusive: bool,
+    ) -> int:
```

Comment:
I assume you copy from the original code without modification? If so, I will skip reviewing this.